### PR TITLE
fix: Guard resolve_css_var() against non-string input from wp_get_global_styles()

### PR DESCRIPTION
**Pull request by pierorocca**
Related: WOOPMNT-6032

Fixes a fatal `TypeError` in `WC_Payments_Styles_Cache::resolve_css_var()` when `wp_get_global_styles()` returns non-string values (e.g., arrays for `fontFamily` in theme.json v3).

#### Changes proposed in this Pull Request

- Added a `$fallback` parameter to `resolve_css_var()` so non-string or empty input returns the caller-specified default instead of crashing
- Updated all call sites to pass their intended fallback explicitly (e.g. `'inherit'`, `'#ffffff'`, `'16px'`)
- Added unit test covering arrays, integers, null, booleans, empty strings, and normal string passthrough with fallback

The root cause: newer theme.json specs (v3) can represent `fontFamily` as an array rather than a string. When a block theme uses this format, `wp_get_global_styles()` returns an array, which causes a `TypeError` when passed to the `string`-typed `resolve_css_var()`. The appearance is computed during wp-admin page loads (via `WC_Payments_Admin` localizing `woopayAppearance`) and WooPay session/button initialization.

#### Testing instructions

1. Install and activate the **Assembler** theme (this theme uses array-based `fontFamily` in its theme.json, which is what originally triggered the fatal)
2. Enable WooPay in WooPayments settings
3. Visit any WooPayments admin page (e.g.,**Orders > Order Details** or **Payments > Overview** or **Payments > Settings**) — confirm no fatal error
4. Also verify the storefront checkout page loads without error (WooPay button handler also calls this path)
5. Run the unit test: `vendor/bin/phpunit --filter test_resolve_css_var_returns_fallback_for_non_string_input`

---

<details>
<summary>🔍 Review Report (4-agent parallel review)</summary>

### Overall Verdict: APPROVE

| Dimension | Verdict | Notes |
|-----------|---------|-------|
| Architecture Compliance | ✅ PASS | Change is within styles cache utility, outside payment flow. File placement correct (existing class in `includes/`). |
| Security | ✅ PASS | No security surface change. Method is `private static`, only processes `wp_get_global_styles()` output — no user-controlled input. |
| Performance | ✅ PASS | Single `is_string()` opcode branch. No new allocations, loops, or I/O. Entire path runs at most once per request (cached to WP option). |
| Test Coverage | ✅ PASS | Test covers exact bug scenario (array input), edge cases (int, null, bool), and happy path (string passthrough). Uses `ReflectionMethod` appropriately. |

No required changes. No non-blocking suggestions.

</details>

---

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant.
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

---

🤖 Generated with [Claude Code](https://claude.com/claude-code)
☢️ Powered by [Gamma AI Tools](https://github.a8c.com/Automattic/gamma-ai-tools/)

### CODE REVIEW

File: includes/class-wc-payments-styles-cache.php [hunk 1]
```diff
@@ -498,10 +498,17 @@ private static function is_color_light( string $color ): bool {
...
+	 * values are returned as an empty string so callers fall back gracefully.
+	 *
+	 * @param mixed $value The CSS value, possibly a var() reference.
 	 * @return string The resolved value or the original.
 	 */
-	private static function resolve_css_var( string $value ): string {
+	private static function resolve_css_var( $value ): string {
+		if ( ! is_string( $value ) ) {
+			return '';
+		}
```
[the remark this answers is not in the document]
Comment (pierorocca) [reply includes/class-wc-payments-styles-cache.php]: Fixed. Added a `$fallback` parameter to `resolve_css_var()` so non-string or empty input returns the caller-specified default instead of an empty string. All call sites now pass their intended fallback (e.g. `'inherit'`, `'#ffffff'`, `'16px'`) directly.

**Comment by pierorocca** [comment]
### 💡 Note on `??` vs type-unsafe WordPress returns

The core issue here was a subtle interaction between PHP's `??` (null coalescing) and type-unsafe returns from WordPress APIs like `wp_get_global_styles()`.

`??` only triggers on `null`, not on "wrong type" — so `$array ?? 'fallback'` returns the array, not `'fallback'`. This meant the original call sites like:

```php
$font_family = self::resolve_css_var( $styles['typography']['fontFamily'] ?? 'inherit' );
```

Would pass an array through when `fontFamily` existed as `['System Sans-Serif']` in theme.json v3, causing the fatal error. The initial fix (returning `''` for non-strings) prevented the crash but silently lost the intended `'inherit'` default.

The `$fallback` parameter approach co-locates the "what to do with bad input" decision with the type check, so every call site explicitly declares its default:

```php
$font_family = self::resolve_css_var( $styles['typography']['fontFamily'] ?? '', 'inherit' );
```

**Comment by pierorocca** [comment]
@mgascam added you manually as a review as this fixes a fatal error on wp-admin pages related to https://github.com/Automattic/woocommerce-payments/pull/11427. This needs to be included along with the already merged code or we'll take down sites running Assembler or similar.

**Review by htdat — COMMENTED** [review]
> Fixes a fatal TypeError in WC_Payments_Styles_Cache::resolve_css_var() when wp_get_global_styles() returns non-string values (e.g., arrays for fontFamily in theme.json v3).

This PR could fix this issue; however, I think we can even improve it further by inspecting the non-string values (array) to extract the proper value for theme.json v3. 

@pierorocca - what's the theme and WP version are you testing with? I would like to test it to see what really happens before applying the fix for it.

**Comment by pierorocca** [comment]
> > Fixes a fatal TypeError in WC_Payments_Styles_Cache::resolve_css_var() when wp_get_global_styles() returns non-string values (e.g., arrays for fontFamily in theme.json v3).
> 
> This PR could fix this issue; however, I think we can even improve it further by inspecting the non-string values (array) to extract the proper value for theme.json v3.
> 
> @pierorocca - what's the theme and WP version are you testing with? I would like to test it to see what really happens before applying the fix for it.

Assembler theme per the included test instructions @htdat. I encountered the fatal after my other PR merged into develop and after switching to Assembler. I was trying to replicate the observed issues on WOOPMNT-5907.

**Comment by pierorocca** [comment]
Fixes WOOPMNT-6030

**Comment by pierorocca** [comment]
Also my environment is broke to test any changes @htdat so if you all have an improvement over this working fix please commit it directly.

**Comment by htdat** [comment]
@pierorocca I created the issue ~https://linear.app/a8c/issue/WOOPMNT-5907/~ https://linear.app/a8c/issue/WOOPMNT-6032/
If we do not want to resolve the `ref` value like this `"ref": "styles.elements.heading.typography.fontFamily"` - [code ref](https://github.a8c.com/Automattic/wpcom-themes/blob/7396e81c9b4f7559e034513c0d8abf7356c5557e/assembler/theme.json#L374), your approach is ok. However, I think we would aim to resolve this `ref` value too?

**Comment by htdat** [comment]
FYI. I've pushed out this PR https://github.com/Automattic/woocommerce-payments/pull/11522 where I tried to resolve the `ref` property (with a value such as `styles.elements.heading.typography.fontFamily` too).

**Comment by htdat** [comment]
closing this PR, following up on https://github.com/Automattic/woocommerce-payments/pull/11522
